app/models/report.rb

```diff
@@ -93,6 +93,208 @@ def cleanup_report
+                'checklist_id' => step_checklist.id
+              },
+              'sort_order' => nil,
+              'children' => []
+            }
+          end
+          if step.step_comments.any?
+            step_children << {
+              'type_of' => 'step_comments',
+              'id' => {
+                'step_id' => step.id
+              },
+              'sort_order' => 'asc',
+              'children' => []
+            }
+          end
+
+          module_children << {
+            'type_of' => 'step',
+            'id' => {
+              'step_id' => step.id
+            },
+            'sort_order' => nil,
+            'children' => step_children
+          }
+        end
+
+        my_module.results.each do |result|
+          if result.asset
+            result_comments = []
+
+            if result.result_comments.any?
+              result_comments << {
+                'type_of' => 'result_comments',
+                'id' => {
+                  'result_id' => result.id
+                },
+                'sort_order' => 'asc',
+                'children' => []
+              }
+            end
+
+            module_children << {
+              'type_of' => 'result_asset',
+              'id' => {
+                'result_id' => result.id
+              },
+              'sort_order' => nil,
+              'children' => []
+            }
+          elsif result.table
+            result_comments = []
+
+            if result.result_comments.any?
+              result_comments << {
+                'type_of' => 'result_comments',
+                'id' => {
+                  'result_id' => result.id
+                },
+                'sort_order' => 'asc',
+                'children' => []
+              }
+            end
+
+            module_children << {
+              'type_of' => 'result_table',
+              'id' => {
+                'result_id' => result.id
+              },
+              'sort_order' => nil,
+              'children' => result_comments
+            }
+          elsif result.result_text
+            result_comments = []
+
+            if result.result_comments.any?
+              result_comments << {
+                'type_of' => 'result_comments',
+                'id' => {
+                  'result_id' => result.id
+                },
+                'sort_order' => 'asc',
+                'children' => []
+              }
+            end
+
+            module_children << {
+              'type_of' => 'result_text',
+              'id' => {
+                'result_id' => result.id
+              },
+              'sort_order' => nil,
+              'children' => result_comments
+            }
+          end
+        end
+
+        if my_module.activities.any?
+          module_children << {
+            'type_of' => 'my_module_activity',
+            'id' => {
+              'my_module_id' => my_module.id
+            },
+            'sort_order' => 'asc',
```

Comment:
Seriously, helper method with optional `params` argument would make this so much clearer.